### ISSUE TITLE
FR: Automatic completion of parameters for function properties

### ISSUE THREAD

**Issue by RedBrogdon** [dart-lang/sdk#35884]
(I'm adding this feature request here after discussing it with @scheglov) 

When using some Flutter widgets (like LayoutBuilder), I often reach a place where I need to create a function of a particular type to fill a required property:

```dart
LayoutBuilder(
  builder: // I need to create a function here.
)
```

LayoutBuilder.builder is of type `LayoutWidgetBuilder`, which is:

```dart
Widget Function(BuildContext context, BoxConstraints constraints);
```

I can never remember the names and order of the parameters, and would love it if IntelliJ and the analysis system could auto-fill the declaration for me, either as:

```dart
LayoutBuilder(
  builder: (context, constraints) $CURSOR_HERE$,
)
```

or 

```dart
LayoutBuilder(
  builder: (context, constraints) {
    $CURSOR_HERE$
  },
)
```

**Comment by shinayser** [comment]
@pq this seems the same issue as #40013, right?

**Comment by pq** [comment]
Yes.  Implemented w/ e0c178b.

Thanks for the help w/ triage @shinayser!
